File: INFO1103/tutorial/Triforce.java
import java.util.*;
import java.lang.*;

public class Triforce{
    public static void main(String[] args){
        Scanner keyboard = new Scanner(System.in);
        try{
            System.out.print("Enter height: ");
            int height = keyboard.nextInt();
            System.out.println();
			if(height >= 2 && height <= 20){
                firstLevel(height);
                baseLevel(height);
            }else{
				System.out.println("Invalid height.");
            }
        }catch(InputMismatchException ex){
            System.out.println();
			System.out.println("Invalid height.");
        }
    }
    public static void firstLevel(int input){
        int baseLine = input - 1;
        for(int i = 0; i < input; i++){
            for(int x = input * 2 - 1; x > i; x--){
                System.out.print(" ");
            }
            System.out.print("/");
            for(int j = 0; j < i * 2; j++){
                if(i == baseLine ){
                    System.out.print("_");
                }else{
                System.out.print(" ");
                }
            }
            System.out.println("\\");
        }
        return;
    }
    
    public static void baseLevel(int input){
        int baseLine = input - 1;
        for(int i = 0 ; i < input; i++){
            for(int j = 0; j < input - i - 1; j++){
                System.out.print(" ");
            }
            System.out.print("/");
            for(int x = 0; x < i * 2; x++){
                if(i == baseLine){
                    System.out.print("_");
                }else{
                    System.out.print(" ");
                }
            }
            System.out.print("\\");
            for(int z = 0; z < ( (input - 1) * 2) - (i * 2) ;z++ ){
                System.out.print(" ");
            }
            System.out.print("/");
            for(int x = 0; x < i * 2; x++){
                if(i == baseLine){
                    System.out.print("_");
                }else{
                    System.out.print(" ");
                }
            }
            System.out.print("\\");
            System.out.println();
        }
    }
}

File: COMP2129/Triforce.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

void firstLevel( int height );
void baseLevel( int height );

int main(){
    
    printf("Enter height: ");
    char ch[100];
    scanf("%s", ch);
    printf("\n");
    if ( isalpha(ch[0]) )
    {
        printf("Invalid height.\n");
        return 1;
    }
    int height = atoi( ch );
    if ( height < 2 || height > 20 )
    {
        printf("Invalid height.\n");
        return 1;
    }
        
    firstLevel( height);
    baseLevel( height);

    return 0;
}

void firstLevel( int height )
{
    int baseLine = height - 1;
    for( int i = 0; i < height; i++ )
    {
        for ( int x = height * 2 - 1; x > i; x -- )
        {
            printf(" ");
        }
        printf("/");
        for ( int j = 0; j < i*2; j++)
        {
            if( i == baseLine )
                printf("_");
            else
                printf(" ");
        }
        printf("\\");
        printf("\n");
    }
    return;
}

void baseLevel( int height )
{
    int baseLine = height - 1;
    for( int i = 0; i < height; i++)
    {
        for ( int j = 0; j < height - i - 1;j++)
            printf(" ");
        printf("/");
        for ( int x = 0; x < i * 2; x++)
        {
            if ( i == baseLine )
                printf("_");
            else
                printf(" ");
        }
        printf("\\");
        for ( int z = 0; z <( (height - 1 ) *2) - (i*2); z++)
            printf(" ");
        printf("/");
        for(int x = 0; x < i * 2; x++)
        {
                if ( i == baseLine )
                printf("_");
            else
                printf(" ");
        }
        printf("\\");
        printf("\n");
    }
}